### DISCUSSION TITLE
XDrip doesn't find a G7 paired to a Tandem Slim X2

### DISCUSSION THREAD

**Discussion by Gonzmdv** [NightscoutFoundation/xDrip#3269] [Bug reports]
Hello, I have been using G6 sensors with my Tandem Slim X2, XDrip and my Samsung Galaxy 4 for a long time without any single problem, but yesterday I started a G7 for the first time and XDrip has not been able to find or detect the G7 sensor. I started the sensor using the pump and xDrip gave me one single reading, and that's all.
I'm using a Pixel 6 pro, Android 14 and xDrip 641cc94-2024.01.02

Thank you.

**Comment by Navid200** [comment]
Please post a screenshot of the system status page.

**Comment by Gonzmdv** [comment]
I paired a G6 later because I need the glucose readings in my watch these days while I'm snowboarding. I still have the G7 paired with the pump.

**Reply by Navid200** [reply]
There is not much we can do then.

**Comment by Gonzmdv** [comment]
I will try to reconnect my G7 to xDrip as soon as I finish my Christmas  Holidays and I will send you a Status screenshot.  At the moment, I need to keep my G6 alive. The G7 works pretty bad.

**Reply by Navid200** [reply]
Sure, sounds good

**Comment by Gonzmdv** [comment]
Hello again. I finished my holidays and I'm trying it again. Should I do anything on xDrip to change from my current G6 active session to a new G7 session other than changing the transmitter ID?
The G7 sensor is currently active in the pump.
Thank you.

**Reply by Navid200** [reply]
If you have not scanned the QR code, scan it.  You need to do it only once after installing xDrip.  So, if you have done it already, there is no need.  But, it doesn't hurt.

If you are still using the stable release of xDrip, you should at least update to the current alpha release: https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.11.14

There have been so many changes recently (Nightly releases) to clean up G7 user interface.  But, those are mostly cosmetic and don't affect connectivity.

Before changing the transmitter ID, it is a good idea to delete old G7 transmitter IDs from your phone Bluetooth settings.

If you have old G7 used devices, put them all in the microwave.  I'm not talking about unopened sealed G7 devices.

**Comment by Gonzmdv** [comment]
I always use the latest alpha release. What QR do yo mean? Should I stop in xDrip the running g6 session? I don't want to stop the G6 sensor.

**Reply by Navid200** [reply]
No, you should never stop sensor in xDrip, neither for G6 and definitely not for G7.

I mean the QR code shown on this page: https://navid200.github.io/xDrip/docs/Dexcom/G7.html

**Reply by Navid200** [reply]
Let me correct that.  If you are using a G6 and you want to use a new sensor on the transmitter, you have to stop sensor before you can start a new one on the same transmitter.  But, in this case, as you don't want to stop the G6 sensor, you should not stop sensor.

**Comment by Gonzmdv** [comment]
Then, I should only change the Transmitter ID in settings and wait xDrip to connect to the new G7, true? That's what I have been doing, but xDrip doesn't detect the G7.

**Reply by Navid200** [reply]
You have to be patient.  Let me see a screenshot of the status page: https://navid200.github.io/xDrip/docs/StatusG5G6.html

**Comment by Gonzmdv** [comment]
![Screenshot_20240112-103729](https://github.com/NightscoutFoundation/xDrip/assets/45602955/03f53750-38c7-44a0-ba8c-87b951a9c6d4)

**Reply by Navid200** [reply]
Why is engineering mode enabled?

**Reply by Gonzmdv** [reply]
I don't remember. Could it be that I enabled it when I installed xDrip in the Galaxy Watch?

**Reply by Navid200** [reply]
I don't know.  

You need to disable it before we can proceed.  The following page tells you how.
https://navid200.github.io/xDrip/docs/Engineering-Mode.html

**Reply by Gonzmdv** [reply]
Disabled. How did you know that it was enabled? I have been trying to connect the G7 again. I send you the status page. Thanks again.

![Screenshot_20240112-171225](https://github.com/NightscoutFoundation/xDrip/assets/45602955/e9a73f8e-4dd8-45d0-a654-a8008c232e0d)

**Reply by Navid200** [reply]
There are items on the status page that prove engineering mode is enabled so that those of us who help people who ask to see the status page frequently can see that it is enabled.

It says G7 is more that 9 days old.  Is that right?  So, you inserted it more than 9 days ago?
Do you also have Dexcom app installed?  It counts if it is even on another phone.

**Reply by Gonzmdv** [reply]
Yes, It is more than nine days know. I have the Dexcom G7 app installed, but I stopped it and it is not running. I installed it because I was not able to connect the G7 to xDrip.

**Reply by Navid200** [reply]
What do you mean you stopped it?  How did you stop it?

**Reply by Gonzmdv** [reply]
I don't know the exact setting in English, but you long-press the app icon, App info, Force detention. That way you know the app is not running.

**Reply by Navid200** [reply]
Uninstall it or forget about using xDrip for G7.

**Comment by Gonzmdv** [comment]
Ok, uninstall and try again.

**Reply by Navid200** [reply]
After uninstalling the Dexcom app, restart the phone.

If it does not connect, change the last character of transmitter ID and OK.  Then, change it back.   This will trigger another scan cycle.

**Reply by Gonzmdv** [reply]
Uninstalled, forgot Bluetooth devices (Dexcom), rebooted, changed last character of transmitter ID and changed it back, but cannot connect yet. Any other idea? Thank you.

**Reply by Navid200** [reply]
Look at your t:slim pump to see exactly when it gets a new reading.  After you figure that out, considering there is one reading every 5 minutes, calculate exactly when the next reading is going to be.  Exactly 1 minute before then, go to the classic status page and tap on restart collector.



For anyone else who finds this thread using search, please if you follow our instructions and something doesn't work, please tell us right away then.
Please don't go install Dexcom app and then tell us.  Please don't do that.

**Reply by Gonzmdv** [reply]
I doesn't work. However, as soon as I introduce the G6 transmitter ID I get readings. I have also been getting readings from the G6 while the Dexcom G7 app was getting readings from the G7, both on the same phone.

**Reply by Navid200** [reply]
The G7 collector is new.  We will be improving it.  At least, I hope.

After it gets messed up, it will take a lot of patience to get it going.
One guaranteed solution, is to uninstall xDrip and do a fresh install.  But, that will result in you losing all the data history and settings in xDrip.
Considering you are almost done with this G7, you might as well wait for the next one.

If you have another phone, you can install xDrip on it, scan the QR code, do NOT install Dexcom app, enter the pairing code, and see how easy it is for it to connect.

When the time comes for the next device, after you insert the device, and enter the pairing code in your t:slim pump and get readings on it, just enter the pairing code in xDrip, and you will get readings there.

Of course, if you install Dexcom app and connect, the phone will pair and it will behave as it does not after you switch from Dexcom app to xDrip.

If you really want to get it going, you will need to restart the phone and wait for it to scan.  If it does not find it, tap on restart  collector to keep the scanner going until it connects.

**Reply by Gonzmdv** [reply]
Ok, thank you very much, let's see tomorrow with the new G7. I will also try in my old Nexus.

**Comment by Gonzmdv** [comment]
I tried the current G7 in my old phone. First updated xDrip. It doesn't work either. Sensor status: Missing QR Code. What does it mean?

![Screenshot_20240113-000254](https://github.com/NightscoutFoundation/xDrip/assets/45602955/b939d0f2-f946-4351-b26c-49cc352f1439)

**Reply by Navid200** [reply]
It means you haven't scanned the QR code.

**Comment by Gonzmdv** [comment]
What QR Code, the one on the G7 sensor?

**Reply by Navid200** [reply]
No, the one on the instructions page: https://navid200.github.io/xDrip/docs/Dexcom/G7.html

**Reply by Gonzmdv** [reply]
Sorry, I didn't know there was an instructions page. What is that code for?

**Reply by Navid200** [reply]
It is for letting you use xDrip to collect from a G7.

**Comment by Gonzmdv** [comment]
Followed the instructions in my old phone and now is working with the G7 I inserted a couple of hours ago. Sorry, I didn't know anything about that instructions and scanning the code. Sorry for wasting your time. Is it necessary to uninstall and reinstall xDrip? I don't want to configure the alarms again.

**Reply by Navid200** [reply]
No, there is no need to uninstall.
If you follow the instructions, it will work.

**Comment by Gonzmdv** [comment]
Yes, It works fine now. Why the need to scan a code? Isn't it posible to activate the needed options? After scanning for G7 use, can I still use G6  sensors? Thank you.

**Reply by Navid200** [reply]
It's a requirement for using xDrip.  I use G7 with xDrip and I scan the code and I am happy with it.

You can either use G6 or G7.  You cannot use both from the same xDrip app.
If you want to change the xDrip hardware data source, you always can.  Using one does not lock you to it.  But, as I stated already, you cannot use two different hardware data sources at once in the same xDrip install.
